src/utils/cosmosdb_model.ts

```diff
@@ -211,6 +212,56 @@ export abstract class CosmosdbModel<
     )(newDocument, options);
   }
 
+  /**
+   * Partial update an existing document identified by searchKey.
+   *
+   * @param searchKey the document identifier
+   * @param partialDocument the field to update
+   * @param options the request option (not required)
+   * @returns a task either containig the updated document
+   */
+  public patch(
+    searchKey: DocumentSearchKey<TR, CosmosDocumentIdKey, PartitionKey>,
+    partialDocument: Partial<T>,
+    options?: RequestOptions
+  ): TE.TaskEither<CosmosErrors, TR> {
+    // documentId must be always valued,
+    // meanwhile partitionKey might be undefined
+    const documentId = searchKey[0];
+    const partitionKey = searchKey[1] || documentId;
+    return pipe(
+      partialDocument,
+      R.toArray(),
+      RA.map(entry => ({
+        op: PatchOperationType.add,
```

Comment:
what if we have to update an existing path? We can safely use `PatchOperationType.add` as op ?

Reply:
Yes, add operation replace existing value:
```
 Add performs one of the following, depending on the target path:
    If the target path specifies an element that does not exist, it is added.
    If the target path specifies an element that already exists, its value is replaced.
    If the target path is a valid array index, a new element will be inserted into the array at the specified index. It shifts existing elements to the right.
    If the index specified is equal to the length of the array, it will append an element to the array. Instead of specifying an index, you can also use the - character. It will also result in the element being appended to the array.
```